### PR TITLE
partition: global index GA

### PR DESCRIPTION
**Pull request by hfxsd**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/18659
- Other reference link(s):https://github.com/pingcap/docs/pull/18543

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: partitioned-table.md [hunk 1]
```diff
@@ -1789,6 +1785,7 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
...
 - DDL operations such as `ADD PARTITION`, `DROP PARTITION`, `TRUNCATE PARTITION`, `REORGANIZE PARTITION`, `SPLIT PARTITION`, and `EXCHANGE PARTITION` also trigger updates to global indexes. The results of these DDL operations will only be returned after the global indexes of the corresponding tables are fully updated. This can delay operations that usually require quick DDL completion, such as data archiving operations (`EXCHANGE PARTITION`, `TRUNCATE PARTITION`, and `DROP PARTITION`). In contrast, when global indexes are not involved, these DDL operations can be completed immediately.
 - By default, the primary key of a partitioned table is a clustered index and must include the partition key. If you require the primary key to exclude the partition key, you can explicitly specify the primary key as a non-clustered global index when creating the table, for example, `PRIMARY KEY(col1, col2) NONCLUSTERED GLOBAL`.
+- If a global index is added to an expression column, you need to collect statistics manually for that global index.
```
Comment (Defined2014) [inline partitioned-table.md:1788]:
```suggestion
- If a global index is added to an expression column or a prefix column, you need to collect statistics manually for that global index.
```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md (hunk 1 above)
Comment (Frank945946) [reply partitioned-table.md:1788]: If a global index is added to an expression column or if the global index is also a prefix index, you need to manually collect statistics for that global index.

Comment (hfxsd) [inline partitioned-table.md:1788]:
```suggestion
- If a global index is added to an expression column, or a prefix index followed by `GLOBAL` becomes a global index ( for example, `INDEX idx1 (a(3)) GLOBAL`), you need to collect statistics manually for that global index.
```

**Review by mjonss — APPROVED** [review]
LGTM, but the testing for GA is not completed yet. Also EXCHANGE PARTITION is not supported with Global Index, and most likely will not in 8.4 timeframe.

File: partitioned-table.md [hunk 2]
```diff
@@ -1789,6 +1785,7 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
...
 - DDL operations such as `ADD PARTITION`, `DROP PARTITION`, `TRUNCATE PARTITION`, `REORGANIZE PARTITION`, `SPLIT PARTITION`, and `EXCHANGE PARTITION` also trigger updates to global indexes. The results of these DDL operations will only be returned after the global indexes of the corresponding tables are fully updated. This can delay operations that usually require quick DDL completion, such as data archiving operations (`EXCHANGE PARTITION`, `TRUNCATE PARTITION`, and `DROP PARTITION`). In contrast, when global indexes are not involved, these DDL operations can be completed immediately.
 - By default, the primary key of a partitioned table is a clustered index and must include the partition key. If you require the primary key to exclude the partition key, you can explicitly specify the primary key as a non-clustered global index when creating the table, for example, `PRIMARY KEY(col1, col2) NONCLUSTERED GLOBAL`.
+- If a global index is added to an expression column, or a global index is also a prefix index (for example `UNIQUE KEY idx_email_prefix (email(10)) GLOBAL)`), you need to collect statistics manually for that global index.
```
Comment (hfxsd) [inline partitioned-table.md:1788]:
```suggestion
- If a global index is added to an expression column, or a global index is also a prefix index (for example `UNIQUE KEY idx_id_prefix (id(10)) GLOBAL)`), you need to collect statistics manually for that global index.
```

**Review by Frank945946 — APPROVED** [review]

File: partitioned-table.md [hunk 3]
```diff
@@ -1789,6 +1785,7 @@ ALTER TABLE t1 PARTITION BY HASH (col1) PARTITIONS 3 UPDATE INDEXES (uidx12 LOCA
...
 - DDL operations such as `ADD PARTITION`, `DROP PARTITION`, `TRUNCATE PARTITION`, `REORGANIZE PARTITION`, `SPLIT PARTITION`, and `EXCHANGE PARTITION` also trigger updates to global indexes. The results of these DDL operations will only be returned after the global indexes of the corresponding tables are fully updated. This can delay operations that usually require quick DDL completion, such as data archiving operations (`EXCHANGE PARTITION`, `TRUNCATE PARTITION`, and `DROP PARTITION`). In contrast, when global indexes are not involved, these DDL operations can be completed immediately.
 - By default, the primary key of a partitioned table is a clustered index and must include the partition key. If you require the primary key to exclude the partition key, you can explicitly specify the primary key as a non-clustered global index when creating the table, for example, `PRIMARY KEY(col1, col2) NONCLUSTERED GLOBAL`.
+- If a global index is added to an expression column, or a global index is also a prefix index (for example `UNIQUE KEY idx_id_prefix (id(10)) GLOBAL`), you need to collect statistics manually for that global index.
```
Comment (lilin90) [inline partitioned-table.md:1788]:
```suggestion
- If a global index is added to an expression column, or a global index is also a prefix index (for example `UNIQUE KEY idx_id_prefix (id(10)) GLOBAL`), you need to collect statistics manually for this global index.
```

File: system-variables.md [hunk 4]
```diff
@@ -2147,10 +2147,6 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 
 ### tidb_enable_global_index <span class="version-mark">New in v7.6.0</span>
 
-> **Warning:**
->
-> The feature controlled by this variable is experimental. It is not recommended that you use it in the production environment. This feature might be changed or removed without prior notice. If you find a bug, you can report an [issue](https://github.com/pingcap/tidb/issues) on GitHub.
```
Comment (lilin90) [inline system-variables.md:2152]: This warning is also deleted in https://github.com/pingcap/docs/pull/18984/files.

**Review by lilin90 — APPROVED** [review]
/lgtm

**Review by mjonss — COMMENTED** [review]
Should [basic-features.md `Global Index`](https://docs.pingcap.com/tidb/dev/basic-features#indexing-and-constraints) be updated as well in this PR or will that be handled in another PR?

**Review by Defined2014 — APPROVED** [review]

**Comment by hfxsd** [comment]
> Should [basic-features.md `Global Index`](https://docs.pingcap.com/tidb/dev/basic-features#indexing-and-constraints) be updated as well in this PR or will that be handled in another PR?

Updated in another PR https://github.com/pingcap/docs/pull/18991/files#r1772509629

**Pull request merged**